client/src/pages/AddCourse.jsx

```diff
@@ -0,0 +1,58 @@
+import { useEffect, useState } from 'react';
+import { request } from "../utils"
+import { useAuth } from '../components/AuthProvider';
+
+const AddCourse = () => {
+  const user = useAuth().storedUser;
+  const [courses, setCourses] = useState([])
+  const [clickedCourses, setClickedCourses] = useState([]);
+

```

Comment:
in the future, when you have the user in any component, please check that you have it (i.e. its not null) and redirect to login if otherwise